spring-boot-2-demo-app/src/main/java/io/github/mfvanek/spring/boot2/test/config/WebClientConfig.java

```diff
@@ -0,0 +1,17 @@
+package io.github.mfvanek.spring.boot2.test.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.reactive.function.client.WebClient;
+
+
+@Configuration
+@RequiredArgsConstructor
+public class WebClientConfig {
+
+    @Bean
+    WebClient webClient() {
+        return WebClient.builder().build();

```

Comment:
Let's specify here baseUrl `@Value("${app.external-base-url}")`

Reply:
I placed this string in PublicApiService class, as I need to put url to retry logs. I didn't want to put this string twice. Should I repeat this url string in WebClientConfig class or it is better to remove url from retry logs in PublicApiService class?

Reply:
You have baseUrl to the target service and endpoints within it.
Just concat if you need to get full path